xds/internal/balancer/outlierdetection/subconn_wrapper.go

```diff
@@ -72,3 +94,28 @@ func (scw *subConnWrapper) uneject() {
 func (scw *subConnWrapper) String() string {
 	return fmt.Sprintf("%+v", scw.addresses)
 }
+
+func (scw *subConnWrapper) RegisterHealthListener(listener func(balancer.SubConnState)) {
+	scw.mu.Lock()
+	defer scw.mu.Unlock()
```

Comment:
Nit: we can grab this lock after the check for `scw.healthListenerEnabled`, right?